drivers/storage/portworx/component/securitycontextconstraints.go

```diff
@@ -81,6 +81,35 @@ func (s *scc) crdExists(cluster *opcorev1.StorageCluster) bool {
 	}
 }
 
+func deepEqualSccTemplate(scc, existingScc *ocp_secv1.SecurityContextConstraints) bool {
+	scc.ResourceVersion = existingScc.ResourceVersion
+	if !(scc.Priority == nil && existingScc.Priority == nil ||
+		scc.Priority != nil && existingScc.Priority != nil && *scc.Priority == *existingScc.Priority) ||
+		!(scc.AllowPrivilegeEscalation == nil && existingScc.AllowPrivilegeEscalation == nil ||
+			scc.AllowPrivilegeEscalation != nil && existingScc.AllowPrivilegeEscalation != nil &&
+				*scc.AllowPrivilegeEscalation == *existingScc.AllowPrivilegeEscalation) {
+		return false
+	}
+
+	// Overwrite auto-generated fields to existing scc value to bypass DeepEqual check
+	scc.ObjectMeta.UID = existingScc.ObjectMeta.UID
+	scc.ObjectMeta.Generation = existingScc.ObjectMeta.Generation
+	scc.ObjectMeta.CreationTimestamp = existingScc.ObjectMeta.CreationTimestamp
+	scc.ObjectMeta.ManagedFields = existingScc.ObjectMeta.ManagedFields
+	scc.ObjectMeta.GenerateName = existingScc.ObjectMeta.GenerateName
+	scc.ObjectMeta.Finalizers = existingScc.ObjectMeta.Finalizers
+	scc.ObjectMeta.OwnerReferences = existingScc.ObjectMeta.OwnerReferences
+	scc.TypeMeta.Kind = existingScc.TypeMeta.Kind
+	scc.TypeMeta.APIVersion = existingScc.TypeMeta.APIVersion
+	scc.Priority = existingScc.Priority
+	scc.AllowPrivilegeEscalation = existingScc.AllowPrivilegeEscalation
+	if scc.Groups == nil {
+		// Update will override nil Groups
```

Comment:
this may cause issue if we update groups from non-empty (and non-default) to nil, what's the value that nil is overwritten to, we may need special handling for default value

Reply:
Will use empty slice